### PR TITLE
Bag of mixed candy

### PR DESCRIPTION
**Pull request by jonmv**
@hakonhall please review and merge.

### CODE REVIEW

File: client/go/cmd/testdata/tests/expected-suite.out [hunk 1]
```diff
@@ -1,8 +1,11 @@
-Running my test: .... OK
-Running wrong-bool-value.json: Failed step 1:
-Unexpected value at /root/coverage/full: true
+My test: .... OK
+wrong-bool-value.json: failed
+Step 1:
```
Comment (hakonhall) [inline client/go/cmd/testdata/tests/expected-suite.out:3]: `In step 1`  may be better?

File: client/go/cmd/testdata/tests/expected-suite.out [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+Unexpected value at /root/coverage/full
```
Comment (hakonhall) [inline client/go/cmd/testdata/tests/expected-suite.out:4]: Unexpected value **in response** at ...etc ?

**Review by hakonhall — APPROVED** [review]

File: client/go/cmd/testdata/tests/expected-suite.out (hunk 1 above)
Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:3]: Hmm, yes, but it doesn't really match `<step name>"` ...

File: client/go/cmd/testdata/tests/expected-suite.out (hunk 2 above)
Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: Reponse has unexpected value at ... ?

Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: Otherwise the "unexpeccted value" becomes too detached from the relevant path.

Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: But that's just wordy, again ...

Comment (hakonhall) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: LGTM

Comment (hakonhall) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: Or perhaps `In step 1` without name, and `In step 1: <name>` with name?  That way people also learn what the step number means.

Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: That's a great idea!

Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: Does it make sense to do this with files, too?

Comment (hakonhall) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: How would that look?

Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: ```feed-and-search-test.json: Test feeding and searching basics: ... OK```

Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: Perhaps too much.

Comment (hakonhall) [reply client/go/cmd/testdata/tests/expected-suite.out:4]: I suggest the stem of the filename **is** the test name, which avoids the duality and possibly duplicated prefix (<filename>:<testname>).  But we can revisit this later.

**Pull request merged**